proto/cosmwasm/wasm/v1/authz.proto

```diff
@@ -0,0 +1,82 @@
+syntax = "proto3";
+package cosmwasm.wasm.v1;
+
+import "gogoproto/gogo.proto";
+import "cosmos_proto/cosmos.proto";
+import "cosmos/base/v1beta1/coin.proto";
+
+option go_package = "github.com/CosmWasm/wasmd/x/wasm/types";
+
+// ContractExecutionAuthorization defines authorization for wasm execute.
+message ContractExecutionAuthorization {
+  option (cosmos_proto.implements_interface) = "Authorization";
+
+  repeated ContractExecutionGrant grants = 1 [ (gogoproto.nullable) = false ];
+
+  // ContractExecutionGrant a granted execute permission for a single contract
+  message ContractExecutionGrant {
+    // Contract is the address of the smart contract
+    string contract = 1;
+
+    // ExecutionLimit specifies number of executions or spendable amounts
+    oneof execution_limit {
+      InfiniteCalls infinite_calls = 2;
```

Comment:
Do we really need this? I would prefer to not provide Infinite(s) but have high MaxCalls instead. Less options for the CLI or other places that display state.